### ISSUE TITLE
dispExample environment does not respect the minted language key

### ISSUE THREAD

**Issue by yudai-nkt** [T-F-S/tcolorbox#24]
Since [the original post][orig_post] on TeX.SX has been inactive for a while and TeX Live 2017 pretest is going to begin, I'd like to let you know this issue here as well.

Consider the following MWE:

```latex
\documentclass{article}
\usepackage[all]{tcolorbox}
\tcbset{listing engine=minted}
\begin{document}
\begin{tcblisting}{minted language=text}
  \begin{equation}
    \int_{-\infty}^{\infty} \exp (-x^2) \, dx = \sqrt{\pi}
  \end{equation}
\end{tcblisting}
\begin{dispExample}
  \begin{equation}
    \int_{-\infty}^{\infty} \exp (-x^2) \, dx = \sqrt{\pi}
  \end{equation}
\end{dispExample}
\tcbset{
  docexample/.add style={}{
    fontlower=\normalsize,
    minted language=text,
    documentation minted options={
      autogobble=true,
      fontsize=\small
    }
  }
}
\begin{dispExample}
  \begin{equation}
    \int_{-\infty}^{\infty} \exp (-x^2) \, dx = \sqrt{\pi}
  \end{equation}
\end{dispExample}
\end{document}
```

Typesetting this with `pdflatex -shell-escape` produces

[![output of the MWE][mwe]][mwe]

The option `minted language=text` has no effect on the latter `dispExample` environment while other `.add style`'d options do.

Please note that I'm not sure if this issue lies in the `tcolorbox` package.

[orig_post]: https://tex.stackexchange.com/questions/345990/how-to-disable-minted-highlighting-in-tcolorboxs-dispexample-environment
[mwe]: https://i.stack.imgur.com/nqGFn.png

**Comment by T-F-S** [comment]
Currently, the `documentation` library assumes all (minted) texts to be of `latex` language. Therefore, `minted language` has deliberately no effect on `dispExample` environments.

Nevertheless, I take it as a feature request to make this configurable in the same way as `documentation minted options`.

The following patch introduces a new option `documentation minted language` which will be integrated into `tcolorbox` version `4.03` and higher:

```latex
\documentclass{article}
\usepackage[all]{tcolorbox}
\begin{document}

\makeatletter

\tcbset{
  documentation minted language/.store in=\kvtcb@docminted@language,
  documentation minted language=latex,
}

\def\tcb@minted@doc@usetemplisting{%
  \tcb@minted@input@listing{\kvtcb@docmintoptions}{\kvtcb@docminted@language}{\kvtcb@tempfile}{\kvtcb@docmintstyle}%
}

\tcbset{listing engine=minted}

\makeatother

\begin{tcblisting}{minted language=text}
  \begin{equation}
    \int_{-\infty}^{\infty} \exp (-x^2) \, dx = \sqrt{\pi}
  \end{equation}
\end{tcblisting}

\begin{dispExample}
  \begin{equation}
    \int_{-\infty}^{\infty} \exp (-x^2) \, dx = \sqrt{\pi}
  \end{equation}
\end{dispExample}

\tcbset{
  docexample/.add style={}{
    fontlower=\normalsize,
    documentation minted language=text,
    documentation minted options={
      autogobble=true,
      fontsize=\small
    }
  }
}

\begin{dispExample}
  \begin{equation}
    \int_{-\infty}^{\infty} \exp (-x^2) \, dx = \sqrt{\pi}
  \end{equation}
\end{dispExample}

\end{document}
```
![mintedlanguage](https://cloud.githubusercontent.com/assets/7982989/25335657/3fa0727a-28f4-11e7-9578-13f6cc1d53f3.png)

**Comment by yudai-nkt** [comment]
> Currently, the `documentation` library assumes all (minted) texts to be of `latex` language.

I didn't notice that the language option is hard-coded as I only checked `tcbdocumentation.code.tex`.

Thanks again for adding the flexibility!

**Comment by T-F-S** [comment]
Option `documentation minted language` officially implemented with version `4.03`.
